### PR TITLE
Add a banner line for latest hints

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Fixes https://github.com/readthedocs/readthedocs.org/issues/6096

### CODE REVIEW

**Comment by arron-tij** [comment]
Wouldn't it be much less redundant and crisp if it just stated
This banner has a text with a link redirecting the users to the latest **stable** version of your docs.

**Comment by arron-tij** [comment]
Wouldn't it be less redundant and more crisp if it just stated
This banner has a text with a link redirecting the users to the latest **stable** version of your docs.
Also I think it doesn't fix the second part of the issue. Should I go ahead and make another PR @stsewd or this issue has already been assigned?

**Comment by tapaswenipathak** [comment]
@arron-tij: Suggestion - never spam PRs if you aren't a maintainer, write notes on tickets.

**Comment by tapaswenipathak** [comment]
@arron-tij: Suggestion - always write notes on ticket, never spam or write notes on any PRs if you aren't a maintainer. Feel free writing a PR, nothing is assigned, I just felt like writing the tiny PR.

**Comment by stsewd** [comment]
@arron-tij we don't assign issues, but if there is an open PR, that means someone else is working on that issue. You can take the work if the PR is stale and the author hasn't update it in a long time.

Feel free to ask questions on our gitter channel https://gitter.im/rtfd/readthedocs.org

**Comment by tapaswenipathak** [comment]
@stsewd: Please see now.

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ class Project(models.Model):
     show_version_warning = models.BooleanField(
         _('Show version warning'),
         default=False,
-        help_text=_('Show warning banner in non-stable nor latest versions.'),
+        help_text=_('Show warning banner in non-stable nor stable latest versions.'),
```
Comment (stsewd) [inline readthedocs/projects/models.py]: This change isn't correct. What we need to add here is the banner links to the stable version, and it's shown only on non-stable versions (except for latest).

Also, you should add new migration (we use named migrations), since it modifies the help_text from the model.

Comment (tapaswenipathak) [reply readthedocs/projects/models.py]: I see, I can remove that.

**Comment by tapaswenipathak** [comment]
@stsewd: review ping.

File: readthedocs/projects/migrations/0024_add-show-version-warning.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='project',
             name='show_version_warning',
-            field=models.BooleanField(default=False, help_text='Show warning banner in non-stable nor latest versions.', verbose_name='Show version warning'),
+            field=models.URLField(blank=True, help_text='You are seeing unstable version of the documentation, see the stable versions.', verbose_name='Show version warning'),
```
Comment (stsewd) [inline readthedocs/projects/migrations/0024_add-show-version-warning.py]: There is not need to change the field type. Also, the text doesn't match the usage of this field.

File: readthedocs/projects/models.py [hunk 3]
```diff
@@ -248,10 +248,10 @@ class Project(models.Model):
         default=False,
         help_text='If checked, do not show advertising for this project',
     )
-    show_version_warning = models.BooleanField(
+    show_version_warning = models.URLField(
         _('Show version warning'),
-        default=False,
-        help_text=_('Show warning banner in non-stable nor latest versions.'),
+        blank=True,
+        help_text=_('You are seeing unstable version of the documentation, see the stable versions.'),
```
Comment (stsewd) [inline readthedocs/projects/models.py]: Same, you shouldn't change the defaults

File: readthedocs/projects/migrations/0024_add-show-version-warning.py (hunk 2 above)
Comment (tapaswenipathak) [reply readthedocs/projects/migrations/0024_add-show-version-warning.py]: I don't know what you are saying. How do you think you will have a link shown in the banner?

File: readthedocs/projects/models.py (hunk 3 above)
Comment (tapaswenipathak) [reply readthedocs/projects/models.py]: text?

**Comment by tapaswenipathak** [comment]
@stsewd: I did few text fixes, lemme know.

**Comment by tapaswenipathak** [comment]
@stsewd: Can you tell me the correct text? Ref: https://github.com/readthedocs/readthedocs.org/pull/6143#discussion_r322485469.

**Comment by stsewd** [comment]
Something like

> Show a warning banner with a link to the stable version on not stable versions, except for latest

**Comment by tapaswenipathak** [comment]
@stsewd: review ping.

File: readthedocs/projects/migrations/0024_add-show-version-warning.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='project',
             name='show_version_warning',
-            field=models.BooleanField(default=False, help_text='Show warning banner in non-stable nor latest versions.', verbose_name='Show version warning'),
+            field=models.BooleanField(default=False, help_text='Show a warning banner with a link to the stable version on not stable versions, except for latest', verbose_name='Show version warning'),
```
Comment (stsewd) [inline readthedocs/projects/migrations/0024_add-show-version-warning.py]: You shouldn't modify existing migrations

File: readthedocs/projects/migrations/0045_auto_20191017_0241.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (stsewd) [inline readthedocs/projects/migrations/0045_auto_20191017_0241.py]: This should be a named migration

```
python manage.py makemigrations --name update-show-version-warning-help-text
```

File: readthedocs/projects/models.py [hunk 6]
```diff
@@ -251,7 +251,8 @@ class Project(models.Model):
     show_version_warning = models.BooleanField(
         _('Show version warning'),
         default=False,
-        help_text=_('Show warning banner in non-stable nor latest versions.'),
+        help_text=_('Show a warning banner with a link to the stable version on not'
```
Comment (stsewd) [inline readthedocs/projects/models.py]: You could warp this like

```python
help_text=_(
    'Show...'
)
```

This is the style of most of the code

File: docs/versions.rst [hunk 7]
```diff
@@ -62,7 +62,7 @@ Version warning
 ---------------
 
 This is a banner that appears on the top of every page of your docs that aren't stable or latest.
-This banner has a text with a link redirecting the users to the latest version of your docs.
+This banner has a text with a link redirecting the users to the stable version of your docs.
```
Comment (humitos) [inline docs/versions.rst]: This may still be confusing. I'd like to say something like

```suggestion
This banner has a text with a link redirecting the users to the latest stable version of your docs.
```

but with a clearer English sentence.

Comment (humitos) [reply docs/versions.rst]: Mmm... does the banner links to `v0.x` or to `stable`?

File: readthedocs/projects/models.py [hunk 8]
```diff
@@ -251,7 +251,10 @@ class Project(models.Model):
     show_version_warning = models.BooleanField(
         _('Show version warning'),
         default=False,
-        help_text=_('Show warning banner in non-stable nor latest versions.'),
+        help_text=_(
+            'Show a warning banner with a link to the stable version on not'
+            ' stable versions, except for latest.'
```
Comment (humitos) [inline readthedocs/projects/models.py]: Here also. I'd like to improve this sentence. It's really hard to parse and understand the last part: "the stable version on not stable versions, except for latest.". After reading it three times I think I understand what it means, but it was hard to get it.

Maybe splitting it in multiple sentences like:

    Show a warning banner with a link to the stable version.
    The banner is shown in all the versions, but in latest and stable.

File: docs/versions.rst (hunk 7 above)
Comment (stsewd) [reply docs/versions.rst]: We only have one stable version, there isn't something like `latest stable`.

Comment (humitos) [reply docs/versions.rst]: I realized that and that's why my second question.
